src/test/java/com/fasterxml/jackson/core/io/BufferRecyclerPoolTest.java

```diff
@@ -41,7 +45,7 @@ private void checkBufferRecyclerPoolImpl(BufferRecyclerPool pool, boolean checkP
 
         if (checkPooledResource) {
             // acquire the pooled BufferRecycler again and check if it is the same instance used before
-            BufferRecycler pooledBufferRecycler = pool.acquireBufferRecycler();
```

Comment:
Any chance we could verify life-cycle without directly calling internal method(s)?